src/Admin.js

```diff
@@ -2,43 +2,70 @@ import Hotel from './Hotel'
 
 class Admin extends Hotel {
   constructor(customerData, roomsData, bookingsData, currentDate) {
-    super(customerData, roomsData, bookingsData)
+    super(customerData, roomsData, bookingsData, currentDate)
     this.customer = customerData;
     this.rooms = roomsData;
     this.bookings = bookingsData;
     this.date = currentDate;
   }
 
-  getTotalRoomsAvailableToday() {
-
+  getTotalRoomsAvailableToday(date) {
+    let roomsWithCustomers = this.bookings.reduce((acc, room) => {
+      if (date === room.date) {
+        acc.push(room);
+      }
+      return acc;
+    }, []).length;
+    return Number(this.rooms.length - roomsWithCustomers)
   }
 
-  getTotalRevenueToday() {
-
+  getTotalRevenueToday(date) {
+    let bookingsByDate = this.bookings.filter(booking => booking.date === date);
+    let totalRev = bookingsByDate.reduce((acc, room) => {
+      this.rooms.forEach(item => {
+        if(item.number === room.roomNumber)
+          acc += item.costPerNight
+        })
+        return acc
+    }, 0);
+    return Number(totalRev.toFixed(2))
   }
 
-  getPercentRoomsOccupiedToday() {
-
+  getPercentRoomsOccupiedByDate(date) {
+    let bookingsByDate = this.bookings.filter(booking => booking.date === date);
+    let percentOccupied = Number((bookingsByDate.length / 50).toFixed(2))
```

Comment:
Totally works for this, though perhaps think about making the divisor more dynamic?  Maybe a room is under construction, so it'd be dividing by 49 instead of 50?  Low probability, so this is definitely clean for the available data.

Reply:
Good point.